applications/asset_tracker/src/main.c

```diff
@@ -186,14 +207,105 @@ static void shutdown_modem(void)
 #endif
 }
 
+#if defined(CONFIG_LWM2M_CARRIER)
+int lwm2m_carrier_event_handler(const lwm2m_carrier_event_t *event)
+{
+	switch (event->type) {
+	case LWM2M_CARRIER_EVENT_BSDLIB_INIT:
+		LOG_DBG("LWM2M_CARRIER_EVENT_BSDLIB_INIT");
+		k_sem_give(&bsdlib_initialized);
+		break;
+	case LWM2M_CARRIER_EVENT_CONNECTING:
+		LOG_DBG("LWM2M_CARRIER_EVENT_CONNECTING\n");
+		break;
+	case LWM2M_CARRIER_EVENT_CONNECTED:
+		LOG_DBG("LWM2M_CARRIER_EVENT_CONNECTED");
+		k_sem_give(&lte_connected);
+		break;
+	case LWM2M_CARRIER_EVENT_DISCONNECTING:
+		LOG_DBG("LWM2M_CARRIER_EVENT_DISCONNECTING");
+		break;
+	case LWM2M_CARRIER_EVENT_BOOTSTRAPPED:
+		LOG_DBG("LWM2M_CARRIER_EVENT_BOOTSTRAPPED");
+		break;
+	case LWM2M_CARRIER_EVENT_DISCONNECTED:
+		LOG_DBG("LWM2M_CARRIER_EVENT_DISCONNECTED");
+		break;
+	case LWM2M_CARRIER_EVENT_READY:
+		LOG_DBG("LWM2M_CARRIER_EVENT_READY");
+		k_sem_give(&cloud_ready_to_connect);
```

Comment:
@stig-bjorlykke Can you comment if we can safely move this line (allow connection to cloud) to the `LWM2M_CARRIER_EVENT_BOOTSTRAPPED` case?

Reply:
`LWM2M_CARRIER_EVENT_BOOTSTRAPPED` will only be given the first time a device boots and when the SIM is changed, and this event is mostly used in test environments. After the initial device boot the application will only get `LWM2M_CARRIER_EVENT_READY`, and this should happen pretty quick after startup.

The delay between `LWM2M_CARRIER_EVENT_BOOTSTRAPPED` and `LWM2M_CARRIER_EVENT_READY` is carrier dependent, and for some carriers this delay is very small.

Some background: After `LWM2M_CARRIER_EVENT_BOOTSTRAPPED` the library will continue to connect to other servers using DTLS. If moving the "cloud_ready_to_connect" handling it's more likely that the application connect and the library connect will happen at the same time, and because the modem only can handle one handshake at a time one of this connects may fail and you will get a delayed connect anyway.